modules/apps/transfer/keeper/relay.go

```diff
@@ -282,6 +283,24 @@ func (k Keeper) OnRecvPacket(ctx sdk.Context, packet channeltypes.Packet, data t
 		return err
 	}
 
+	if !k.bankKeeper.HasDenomMetaData(ctx, voucherDenom) {
+		metadata := banktypes.Metadata{
+			Description: getMetaDataDescription(denomTrace),
+			DenomUnits: []*banktypes.DenomUnit{
+				{
+					Denom:    denomTrace.BaseDenom,
```

Comment:
VoucherDenom is the IBCHash, so wouldn't it better to have a readable for DenomUnits? Maybe we can use the full path here if needed. Instead of having a seperate hash for each unit we will have paths like...
`
{
    Denom: transfer/channel-10/foobar
    Exponent: 6
},
{
    Denom: transfer/channel-10/ufoobar
    Exponent: 0
}
`